<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>GViz Diagrams</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="GViz-Diagrams_files/libs/clipboard/clipboard.min.js"></script>
<script src="GViz-Diagrams_files/libs/quarto-html/quarto.js"></script>
<script src="GViz-Diagrams_files/libs/quarto-html/popper.min.js"></script>
<script src="GViz-Diagrams_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="GViz-Diagrams_files/libs/quarto-html/anchor.min.js"></script>
<link href="GViz-Diagrams_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="GViz-Diagrams_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="GViz-Diagrams_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="GViz-Diagrams_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="GViz-Diagrams_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">GViz Diagrams</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="gviz-diagrams" class="level1">
<h1>GViz Diagrams</h1>
<p>This code will generate GViz Diagrams for GOIs that we will discuss in the report</p>
<section id="libraries" class="level2">
<h2 class="anchored" data-anchor-id="libraries">Libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>here() starts at C:/R Projects/ZB3101/Assignment 2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.2     ✔ tibble    3.2.1
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.0.4     </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Gviz)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors
Loading required package: stats4
Loading required package: BiocGenerics
Loading required package: generics

Attaching package: 'generics'

The following object is masked from 'package:lubridate':

    as.difftime

The following object is masked from 'package:dplyr':

    explain

The following objects are masked from 'package:base':

    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,
    setequal, union


Attaching package: 'BiocGenerics'

The following object is masked from 'package:dplyr':

    combine

The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs

The following objects are masked from 'package:base':

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,
    unsplit, which.max, which.min


Attaching package: 'S4Vectors'

The following objects are masked from 'package:lubridate':

    second, second&lt;-

The following objects are masked from 'package:dplyr':

    first, rename

The following object is masked from 'package:tidyr':

    expand

The following object is masked from 'package:utils':

    findMatches

The following objects are masked from 'package:base':

    expand.grid, I, unname

Loading required package: IRanges

Attaching package: 'IRanges'

The following object is masked from 'package:lubridate':

    %within%

The following objects are masked from 'package:dplyr':

    collapse, desc, slice

The following object is masked from 'package:purrr':

    reduce

The following object is masked from 'package:grDevices':

    windows

Loading required package: GenomicRanges
Loading required package: GenomeInfoDb
Loading required package: grid</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ensembldb)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicFeatures
Loading required package: AnnotationDbi
Loading required package: Biobase
Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.


Attaching package: 'AnnotationDbi'

The following object is masked from 'package:dplyr':

    select

Loading required package: AnnotationFilter

Attaching package: 'AnnotationFilter'

The following object is masked from 'package:Gviz':

    feature


Attaching package: 'ensembldb'

The following object is masked from 'package:dplyr':

    filter

The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationHub)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocFileCache
Loading required package: dbplyr

Attaching package: 'dbplyr'

The following objects are masked from 'package:dplyr':

    ident, sql


Attaching package: 'AnnotationHub'

The following object is masked from 'package:Biobase':

    cache</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggthemes)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'scales'

The following object is masked from 'package:purrr':

    discard

The following object is masked from 'package:readr':

    col_factor</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(colorspace)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'colorspace'

The following object is masked from 'package:Gviz':

    coords</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/

Please cite:

Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.
clusterProfiler: an R package for comparing biological themes among
gene clusters. OMICS: A Journal of Integrative Biology. 2012,
16(5):284-287

Attaching package: 'clusterProfiler'

The following objects are masked from 'package:ensembldb':

    filter, select

The following object is masked from 'package:AnnotationDbi':

    select

The following object is masked from 'package:IRanges':

    slice

The following object is masked from 'package:S4Vectors':

    rename

The following object is masked from 'package:purrr':

    simplify

The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Mm.eg.db)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(crayon)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'crayon'

The following object is masked from 'package:ggplot2':

    %+%</code></pre>
</div>
</div>
</section>
<section id="variables" class="level2">
<h2 class="anchored" data-anchor-id="variables">Variables</h2>
<p>This chunk declares variables and imports relevant files/data</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>ah <span class="ot">&lt;-</span> <span class="fu">AnnotationHub</span>()</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">query</span>(ah,<span class="fu">c</span>(<span class="st">"GTF"</span>,<span class="st">"GRCm38"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>AnnotationHub with 87 records
# snapshotDate(): 2025-04-08
# $dataprovider: Ensembl
# $species: Mus musculus, mus musculus
# $rdataclass: GRanges
# additional mcols(): taxonomyid, genome, description,
#   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,
#   rdatapath, sourceurl, sourcetype 
# retrieve records with, e.g., 'object[["AH7567"]]' 

            title                                           
  AH7567  | Mus_musculus.GRCm38.70.gtf                      
  AH7628  | Mus_musculus.GRCm38.69.gtf                      
  AH7675  | Mus_musculus.GRCm38.71.gtf                      
  AH7736  | Mus_musculus.GRCm38.72.gtf                      
  AH7799  | Mus_musculus.GRCm38.73.gtf                      
  ...       ...                                             
  AH83746 | Mus_musculus.GRCm38.101.gtf                     
  AH87465 | Mus_musculus.GRCm38.102.abinitio.gtf            
  AH87466 | Mus_musculus.GRCm38.102.chr.gtf                 
  AH87467 | Mus_musculus.GRCm38.102.chr_patch_hapl_scaff.gtf
  AH87468 | Mus_musculus.GRCm38.102.gtf                     </code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>DB <span class="ot">&lt;-</span> <span class="fu">ensDbFromAH</span>(ah[<span class="st">'AH87466'</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Fetching data ... OK
  -------------
Proceeding to create the database.
Processing genes ... </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in ensDbFromGRanges(gff, outfile = outfile, path = path, organism =
orgFromAH, : I'm missing column(s): 'entrezid'. The corresponding database
column(s) will be empty!</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code> Attribute availability:
  o gene_id ... OK
  o gene_name ... OK
  o entrezid ... Nope
  o gene_biotype ... OK
OK
Processing transcripts ... 
 Attribute availability:
  o transcript_id ... OK
  o gene_id ... OK
  o transcript_biotype ... OK
  o transcript_name ... OK
OK
Processing exons ... OK
Processing chromosomes ... OK
Processing metadata ... OK
Generating index ... OK
  -------------
Verifying validity of the information in the database:
Checking transcripts ... OK
Checking exons ... OK</code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>GRCm38 <span class="ot">&lt;-</span> <span class="fu">EnsDb</span>(DB)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevelsStyle</span>(GRCm38) <span class="ot">&lt;-</span> <span class="st">"UCSC"</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>H3K27me3_wt <span class="ot">&lt;-</span> <span class="fu">BigWigFile</span>(<span class="fu">here</span>(<span class="st">'data'</span>,<span class="st">'chipseq'</span>,<span class="st">'GSM5195453_WT_H3K27me3.bw'</span>))</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>H3K27me3_cko <span class="ot">&lt;-</span> <span class="fu">BigWigFile</span>(<span class="fu">here</span>(<span class="st">'data'</span>,<span class="st">'chipseq'</span>,<span class="st">'GSM5195455_cKO_H3K27me3.bw'</span>))</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>H3K27ac_wt <span class="ot">&lt;-</span> <span class="fu">BigWigFile</span>(<span class="fu">here</span>(<span class="st">'data'</span>,<span class="st">'chipseq'</span>,<span class="st">'GSM5195452_WT_H3K27ac.bw'</span>))</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>H3K27ac_ko <span class="ot">&lt;-</span> <span class="fu">BigWigFile</span>(<span class="fu">here</span>(<span class="st">'data'</span>,<span class="st">'chipseq'</span>,<span class="st">'GSM5195454_cKO_H3K27ac.bw'</span>))</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>K27me3_wt_peaks <span class="ot">&lt;-</span> <span class="fu">import</span>(<span class="fu">here</span>(<span class="st">'data'</span>,<span class="st">'chipseq'</span>,<span class="st">'WT_H3K27me3_peaks.broadPeak'</span>))</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevelsStyle</span>(K27me3_wt_peaks) <span class="ot">&lt;-</span> <span class="st">"UCSC"</span></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>K27me3_cko_peaks <span class="ot">&lt;-</span> <span class="fu">import</span>(<span class="fu">here</span>(<span class="st">'data'</span>,<span class="st">'chipseq'</span>,<span class="st">'cKO_H3K27me3_peaks.broadPeak'</span>))</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevelsStyle</span>(K27me3_cko_peaks) <span class="ot">&lt;-</span> <span class="st">"UCSC"</span></span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>K27ac_wt_peaks <span class="ot">&lt;-</span> <span class="fu">import</span>(<span class="fu">here</span>(<span class="st">'data'</span>,<span class="st">'chipseq'</span>,<span class="st">'WT_H3K27ac_peaks.broadPeak'</span>))</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevelsStyle</span>(K27ac_wt_peaks) <span class="ot">&lt;-</span> <span class="st">"UCSC"</span></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>K27ac_cko_peaks <span class="ot">&lt;-</span> <span class="fu">import</span>(<span class="fu">here</span>(<span class="st">'data'</span>,<span class="st">'chipseq'</span>,<span class="st">'cKO_H3K27ac_peaks.broadPeak'</span>))</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevelsStyle</span>(K27ac_cko_peaks) <span class="ot">&lt;-</span> <span class="st">"UCSC"</span></span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>my_palette <span class="ot">&lt;-</span> <span class="fu">tableau_color_pal</span>(<span class="at">palette=</span><span class="st">'Color Blind'</span>)(<span class="dv">8</span>)</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>my_palette <span class="ot">&lt;-</span> <span class="fu">desaturate</span>(my_palette,.<span class="dv">3</span>)</span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>my_palette <span class="ot">&lt;-</span> <span class="fu">darken</span>(<span class="fu">desaturate</span>(<span class="fu">colorblind_pal</span>()(<span class="dv">8</span>),<span class="fl">0.5</span>),<span class="fl">0.2</span>)[<span class="dv">2</span><span class="sc">:</span><span class="dv">8</span>]</span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>mm10_bands <span class="ot">&lt;-</span> ah[[<span class="st">'AH53180'</span>]] <span class="sc">|&gt;</span> </span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span></span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">'chrom'</span><span class="ot">=</span><span class="st">'seqnames'</span>,</span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a>                <span class="st">'chromStart'</span><span class="ot">=</span><span class="st">'start'</span>,</span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a>                <span class="st">'chromEnd'</span><span class="ot">=</span><span class="st">'end'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>loading from cache</code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>IdeogramTrack <span class="ot">&lt;-</span> <span class="cf">function</span>(chr) {</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>    Gviz<span class="sc">::</span><span class="fu">IdeogramTrack</span>(<span class="fu">as.character</span>(chr),<span class="at">genome=</span><span class="st">'mm10'</span>,<span class="at">bands=</span>mm10_bands)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>  }</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="visualisation" class="level2">
<h2 class="anchored" data-anchor-id="visualisation">Visualisation</h2>
<p>This part is separated into 2 list of genes, one for each of the biological process we have decided to investigate</p>
<section id="muscle-system-process" class="level3">
<h3 class="anchored" data-anchor-id="muscle-system-process">Muscle system process</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co">#GOI</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>goi_vec <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Myh8"</span>, <span class="st">"Myh3"</span>, <span class="st">"Myog"</span>, <span class="st">"Myc"</span>)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#c("ENSMUSG00000030093", "ENSMUSG00000038203", "ENSMUSG00000022484", "ENSMUSG00000050368",  "ENSMUSG00000027168")</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="co">#gene symbols: Hoxa13, Hoxc10, Hoxd10, Wnt7a, Pax6</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (goi_sym <span class="cf">in</span> goi_vec){ <span class="co">#Generate a figure for each gene</span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>  GRCm38 <span class="ot">&lt;-</span> <span class="fu">dropFilter</span>(GRCm38)</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>  goi <span class="ot">&lt;-</span> (<span class="fu">bitr</span>(goi_sym, <span class="at">fromType =</span> <span class="st">"SYMBOL"</span>, <span class="at">toType =</span> <span class="st">"ENSEMBL"</span>, <span class="at">OrgDb =</span> org.Mm.eg.db))</span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>  goi <span class="ot">&lt;-</span> goi<span class="sc">$</span>ENSEMBL</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>  gr <span class="ot">&lt;-</span> GenomicRanges<span class="sc">::</span><span class="fu">reduce</span>(<span class="fu">genes</span>(GRCm38)[goi] <span class="sc">+</span> <span class="dv">5000</span>)</span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">blue</span>(<span class="fu">paste</span>(<span class="st">"Now Showing: "</span>,  goi_sym, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)))</span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a>  gr_filter <span class="ot">&lt;-</span> <span class="fu">GRangesFilter</span>(gr)</span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>  GRCm38 <span class="ot">&lt;-</span> <span class="fu">addFilter</span>(GRCm38, <span class="fu">GRangesFilter</span>(gr))</span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>  k27me3_wt_track <span class="ot">&lt;-</span> <span class="fu">DataTrack</span>(<span class="fu">import.bw</span>(H3K27me3_wt,<span class="at">which=</span>gr)</span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>                               ,<span class="at">type=</span><span class="st">'hist'</span>,<span class="at">window=</span><span class="sc">-</span><span class="dv">1</span>,<span class="at">fill=</span>my_palette[<span class="dv">2</span>],<span class="at">col.histogram=</span>my_palette[<span class="dv">2</span>]</span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>                               ,<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">25</span>)</span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a>                               ,<span class="at">background.title=</span>my_palette[<span class="dv">2</span>]</span>
<span id="cb31-20"><a href="#cb31-20" aria-hidden="true" tabindex="-1"></a>                               ,<span class="at">name=</span><span class="st">"WT me3"</span>)</span>
<span id="cb31-21"><a href="#cb31-21" aria-hidden="true" tabindex="-1"></a>  k27me3_ko_track <span class="ot">&lt;-</span> <span class="fu">DataTrack</span>(<span class="fu">import.bw</span>(H3K27me3_cko,<span class="at">which=</span>gr)</span>
<span id="cb31-22"><a href="#cb31-22" aria-hidden="true" tabindex="-1"></a>                               ,<span class="at">type=</span><span class="st">'hist'</span>,<span class="at">window=</span><span class="sc">-</span><span class="dv">1</span>,<span class="at">fill=</span>my_palette[<span class="dv">3</span>],<span class="at">col.histogram=</span>my_palette[<span class="dv">3</span>]</span>
<span id="cb31-23"><a href="#cb31-23" aria-hidden="true" tabindex="-1"></a>                               ,<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">25</span>)</span>
<span id="cb31-24"><a href="#cb31-24" aria-hidden="true" tabindex="-1"></a>                               ,<span class="at">background.title=</span>my_palette[<span class="dv">3</span>]</span>
<span id="cb31-25"><a href="#cb31-25" aria-hidden="true" tabindex="-1"></a>                               ,<span class="at">name=</span><span class="st">"cKO me3"</span>)</span>
<span id="cb31-26"><a href="#cb31-26" aria-hidden="true" tabindex="-1"></a>  k27ac_wt_track <span class="ot">&lt;-</span> <span class="fu">DataTrack</span>(<span class="fu">import.bw</span>(H3K27ac_wt,<span class="at">which=</span>gr)</span>
<span id="cb31-27"><a href="#cb31-27" aria-hidden="true" tabindex="-1"></a>                              ,<span class="at">type=</span><span class="st">'hist'</span>,<span class="at">window=</span><span class="sc">-</span><span class="dv">1</span>,<span class="at">fill=</span>my_palette[<span class="dv">4</span>],<span class="at">col.histogram=</span>my_palette[<span class="dv">4</span>]</span>
<span id="cb31-28"><a href="#cb31-28" aria-hidden="true" tabindex="-1"></a>                              ,<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">25</span>)</span>
<span id="cb31-29"><a href="#cb31-29" aria-hidden="true" tabindex="-1"></a>                              ,<span class="at">background.title=</span>my_palette[<span class="dv">4</span>]</span>
<span id="cb31-30"><a href="#cb31-30" aria-hidden="true" tabindex="-1"></a>                              ,<span class="at">name=</span><span class="st">"WT ac"</span>)</span>
<span id="cb31-31"><a href="#cb31-31" aria-hidden="true" tabindex="-1"></a>  k27ac_ko_track <span class="ot">&lt;-</span> <span class="fu">DataTrack</span>(<span class="fu">import.bw</span>(H3K27ac_ko,<span class="at">which=</span>gr)</span>
<span id="cb31-32"><a href="#cb31-32" aria-hidden="true" tabindex="-1"></a>                              ,<span class="at">type=</span><span class="st">'hist'</span>,<span class="at">window=</span><span class="sc">-</span><span class="dv">1</span>,<span class="at">fill=</span>my_palette[<span class="dv">5</span>],<span class="at">col.histogram=</span>my_palette[<span class="dv">5</span>]</span>
<span id="cb31-33"><a href="#cb31-33" aria-hidden="true" tabindex="-1"></a>                              ,<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">25</span>)</span>
<span id="cb31-34"><a href="#cb31-34" aria-hidden="true" tabindex="-1"></a>                              ,<span class="at">background.title=</span>my_palette[<span class="dv">5</span>]</span>
<span id="cb31-35"><a href="#cb31-35" aria-hidden="true" tabindex="-1"></a>                              ,<span class="at">name=</span><span class="st">"cKO ac"</span>)</span>
<span id="cb31-36"><a href="#cb31-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-37"><a href="#cb31-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotTracks</span>(<span class="fu">list</span>(<span class="fu">IdeogramTrack</span>(<span class="fu">seqnames</span>(gr))</span>
<span id="cb31-38"><a href="#cb31-38" aria-hidden="true" tabindex="-1"></a>    ,<span class="fu">GenomeAxisTrack</span>(gr)</span>
<span id="cb31-39"><a href="#cb31-39" aria-hidden="true" tabindex="-1"></a>    ,<span class="fu">AnnotationTrack</span>(GenomicRanges<span class="sc">::</span><span class="fu">reduce</span>(<span class="fu">genes</span>(GRCm38)[goi]),</span>
<span id="cb31-40"><a href="#cb31-40" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fill=</span><span class="st">'gray'</span>,<span class="at">name=</span><span class="st">'DEG'</span>,</span>
<span id="cb31-41"><a href="#cb31-41" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fontcolor.title=</span><span class="st">'black'</span>)</span>
<span id="cb31-42"><a href="#cb31-42" aria-hidden="true" tabindex="-1"></a>    ,<span class="fu">GeneRegionTrack</span>(<span class="fu">getGeneRegionTrackForGviz</span>(GRCm38),</span>
<span id="cb31-43"><a href="#cb31-43" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transcriptAnnotation=</span><span class="st">'symbol'</span>,</span>
<span id="cb31-44"><a href="#cb31-44" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fill=</span>my_palette[<span class="dv">1</span>],<span class="at">col=</span>my_palette[<span class="dv">1</span>],<span class="at">col.line=</span><span class="cn">NULL</span>,</span>
<span id="cb31-45"><a href="#cb31-45" aria-hidden="true" tabindex="-1"></a>                    <span class="at">min.height=</span><span class="dv">4</span>,</span>
<span id="cb31-46"><a href="#cb31-46" aria-hidden="true" tabindex="-1"></a>                    <span class="at">background.title=</span>my_palette[<span class="dv">1</span>],</span>
<span id="cb31-47"><a href="#cb31-47" aria-hidden="true" tabindex="-1"></a>                    <span class="at">name=</span><span class="st">"Ensembl"</span>)</span>
<span id="cb31-48"><a href="#cb31-48" aria-hidden="true" tabindex="-1"></a>    ,k27me3_wt_track</span>
<span id="cb31-49"><a href="#cb31-49" aria-hidden="true" tabindex="-1"></a>    ,<span class="fu">AnnotationTrack</span>(K27me3_wt_peaks,<span class="at">chr=</span><span class="fu">as.character</span>(<span class="fu">seqnames</span>(gr)),</span>
<span id="cb31-50"><a href="#cb31-50" aria-hidden="true" tabindex="-1"></a>                     <span class="at">fill=</span>my_palette[<span class="dv">2</span>],<span class="at">col=</span>my_palette[<span class="dv">2</span>],<span class="at">background.title=</span>my_palette[<span class="dv">2</span>])</span>
<span id="cb31-51"><a href="#cb31-51" aria-hidden="true" tabindex="-1"></a>    ,k27me3_ko_track</span>
<span id="cb31-52"><a href="#cb31-52" aria-hidden="true" tabindex="-1"></a>    ,<span class="fu">AnnotationTrack</span>(K27me3_cko_peaks,<span class="at">chr=</span><span class="fu">as.character</span>(<span class="fu">seqnames</span>(gr)),</span>
<span id="cb31-53"><a href="#cb31-53" aria-hidden="true" tabindex="-1"></a>                     <span class="at">fill=</span>my_palette[<span class="dv">3</span>],<span class="at">col=</span>my_palette[<span class="dv">3</span>],<span class="at">background.title=</span>my_palette[<span class="dv">3</span>])</span>
<span id="cb31-54"><a href="#cb31-54" aria-hidden="true" tabindex="-1"></a>    ,k27ac_wt_track</span>
<span id="cb31-55"><a href="#cb31-55" aria-hidden="true" tabindex="-1"></a>    ,<span class="fu">AnnotationTrack</span>(K27ac_wt_peaks,<span class="at">chr=</span><span class="fu">as.character</span>(<span class="fu">seqnames</span>(gr)),</span>
<span id="cb31-56"><a href="#cb31-56" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fill=</span>my_palette[<span class="dv">4</span>],<span class="at">col=</span>my_palette[<span class="dv">4</span>],</span>
<span id="cb31-57"><a href="#cb31-57" aria-hidden="true" tabindex="-1"></a>                    <span class="at">background.title=</span>my_palette[<span class="dv">4</span>])</span>
<span id="cb31-58"><a href="#cb31-58" aria-hidden="true" tabindex="-1"></a>    ,k27ac_ko_track</span>
<span id="cb31-59"><a href="#cb31-59" aria-hidden="true" tabindex="-1"></a>    ,<span class="fu">AnnotationTrack</span>(K27ac_cko_peaks,<span class="at">chr=</span><span class="fu">as.character</span>(<span class="fu">seqnames</span>(gr)),</span>
<span id="cb31-60"><a href="#cb31-60" aria-hidden="true" tabindex="-1"></a>                     <span class="at">fill=</span>my_palette[<span class="dv">5</span>],<span class="at">col=</span>my_palette[<span class="dv">5</span>],<span class="at">background.title=</span>my_palette[<span class="dv">5</span>])</span>
<span id="cb31-61"><a href="#cb31-61" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb31-62"><a href="#cb31-62" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">from=</span><span class="fu">start</span>(gr),<span class="at">to=</span><span class="fu">end</span>(gr)</span>
<span id="cb31-63"><a href="#cb31-63" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb31-64"><a href="#cb31-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-65"><a href="#cb31-65" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Now Showing:  Myh8 \n"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped</code></pre>
</div>
<div class="cell-output-display">
<p><img src="GViz-Diagrams_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Now Showing:  Myh3 \n"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped</code></pre>
</div>
<div class="cell-output-display">
<p><img src="GViz-Diagrams_files/figure-html/unnamed-chunk-3-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Now Showing:  Myog \n"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped</code></pre>
</div>
<div class="cell-output-display">
<p><img src="GViz-Diagrams_files/figure-html/unnamed-chunk-3-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Now Showing:  Myc \n"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped</code></pre>
</div>
<div class="cell-output-display">
<p><img src="GViz-Diagrams_files/figure-html/unnamed-chunk-3-4.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="cell-fate-commitment" class="level3">
<h3 class="anchored" data-anchor-id="cell-fate-commitment">Cell fate commitment</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co">#GOI</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>goi_vec <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Hoxa13"</span>, <span class="st">"Hoxc10"</span>, <span class="st">"Hoxd10"</span>, <span class="st">"Wnt7a"</span>, <span class="st">"Pax6"</span>)</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#c("ENSMUSG00000030093", "ENSMUSG00000038203", "ENSMUSG00000022484", "ENSMUSG00000050368",  "ENSMUSG00000027168")</span></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a><span class="co">#gene symbols: Hoxa13, Hoxc10, Hoxd10, Wnt7a, Pax6</span></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (goi_sym <span class="cf">in</span> goi_vec){ <span class="co">#Generate a figure for each gene</span></span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>  GRCm38 <span class="ot">&lt;-</span> <span class="fu">dropFilter</span>(GRCm38)</span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a>  goi <span class="ot">&lt;-</span> (<span class="fu">bitr</span>(goi_sym, <span class="at">fromType =</span> <span class="st">"SYMBOL"</span>, <span class="at">toType =</span> <span class="st">"ENSEMBL"</span>, <span class="at">OrgDb =</span> org.Mm.eg.db))</span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>  goi <span class="ot">&lt;-</span> goi<span class="sc">$</span>ENSEMBL</span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a>  gr <span class="ot">&lt;-</span> GenomicRanges<span class="sc">::</span><span class="fu">reduce</span>(<span class="fu">genes</span>(GRCm38)[goi] <span class="sc">+</span> <span class="dv">5000</span>)</span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">blue</span>(<span class="fu">paste</span>(<span class="st">"Now Showing: "</span>,  goi_sym, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)))</span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a>  gr_filter <span class="ot">&lt;-</span> <span class="fu">GRangesFilter</span>(gr)</span>
<span id="cb44-14"><a href="#cb44-14" aria-hidden="true" tabindex="-1"></a>  GRCm38 <span class="ot">&lt;-</span> <span class="fu">addFilter</span>(GRCm38, <span class="fu">GRangesFilter</span>(gr))</span>
<span id="cb44-15"><a href="#cb44-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb44-16"><a href="#cb44-16" aria-hidden="true" tabindex="-1"></a>  k27me3_wt_track <span class="ot">&lt;-</span> <span class="fu">DataTrack</span>(<span class="fu">import.bw</span>(H3K27me3_wt,<span class="at">which=</span>gr)</span>
<span id="cb44-17"><a href="#cb44-17" aria-hidden="true" tabindex="-1"></a>                               ,<span class="at">type=</span><span class="st">'hist'</span>,<span class="at">window=</span><span class="sc">-</span><span class="dv">1</span>,<span class="at">fill=</span>my_palette[<span class="dv">2</span>],<span class="at">col.histogram=</span>my_palette[<span class="dv">2</span>]</span>
<span id="cb44-18"><a href="#cb44-18" aria-hidden="true" tabindex="-1"></a>                               ,<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">25</span>)</span>
<span id="cb44-19"><a href="#cb44-19" aria-hidden="true" tabindex="-1"></a>                               ,<span class="at">background.title=</span>my_palette[<span class="dv">2</span>]</span>
<span id="cb44-20"><a href="#cb44-20" aria-hidden="true" tabindex="-1"></a>                               ,<span class="at">name=</span><span class="st">"WT me3"</span>)</span>
<span id="cb44-21"><a href="#cb44-21" aria-hidden="true" tabindex="-1"></a>  k27me3_ko_track <span class="ot">&lt;-</span> <span class="fu">DataTrack</span>(<span class="fu">import.bw</span>(H3K27me3_cko,<span class="at">which=</span>gr)</span>
<span id="cb44-22"><a href="#cb44-22" aria-hidden="true" tabindex="-1"></a>                               ,<span class="at">type=</span><span class="st">'hist'</span>,<span class="at">window=</span><span class="sc">-</span><span class="dv">1</span>,<span class="at">fill=</span>my_palette[<span class="dv">3</span>],<span class="at">col.histogram=</span>my_palette[<span class="dv">3</span>]</span>
<span id="cb44-23"><a href="#cb44-23" aria-hidden="true" tabindex="-1"></a>                               ,<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">25</span>)</span>
<span id="cb44-24"><a href="#cb44-24" aria-hidden="true" tabindex="-1"></a>                               ,<span class="at">background.title=</span>my_palette[<span class="dv">3</span>]</span>
<span id="cb44-25"><a href="#cb44-25" aria-hidden="true" tabindex="-1"></a>                               ,<span class="at">name=</span><span class="st">"cKO me3"</span>)</span>
<span id="cb44-26"><a href="#cb44-26" aria-hidden="true" tabindex="-1"></a>  k27ac_wt_track <span class="ot">&lt;-</span> <span class="fu">DataTrack</span>(<span class="fu">import.bw</span>(H3K27ac_wt,<span class="at">which=</span>gr)</span>
<span id="cb44-27"><a href="#cb44-27" aria-hidden="true" tabindex="-1"></a>                              ,<span class="at">type=</span><span class="st">'hist'</span>,<span class="at">window=</span><span class="sc">-</span><span class="dv">1</span>,<span class="at">fill=</span>my_palette[<span class="dv">4</span>],<span class="at">col.histogram=</span>my_palette[<span class="dv">4</span>]</span>
<span id="cb44-28"><a href="#cb44-28" aria-hidden="true" tabindex="-1"></a>                              ,<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">25</span>)</span>
<span id="cb44-29"><a href="#cb44-29" aria-hidden="true" tabindex="-1"></a>                              ,<span class="at">background.title=</span>my_palette[<span class="dv">4</span>]</span>
<span id="cb44-30"><a href="#cb44-30" aria-hidden="true" tabindex="-1"></a>                              ,<span class="at">name=</span><span class="st">"WT ac"</span>)</span>
<span id="cb44-31"><a href="#cb44-31" aria-hidden="true" tabindex="-1"></a>  k27ac_ko_track <span class="ot">&lt;-</span> <span class="fu">DataTrack</span>(<span class="fu">import.bw</span>(H3K27ac_ko,<span class="at">which=</span>gr)</span>
<span id="cb44-32"><a href="#cb44-32" aria-hidden="true" tabindex="-1"></a>                              ,<span class="at">type=</span><span class="st">'hist'</span>,<span class="at">window=</span><span class="sc">-</span><span class="dv">1</span>,<span class="at">fill=</span>my_palette[<span class="dv">5</span>],<span class="at">col.histogram=</span>my_palette[<span class="dv">5</span>]</span>
<span id="cb44-33"><a href="#cb44-33" aria-hidden="true" tabindex="-1"></a>                              ,<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">25</span>)</span>
<span id="cb44-34"><a href="#cb44-34" aria-hidden="true" tabindex="-1"></a>                              ,<span class="at">background.title=</span>my_palette[<span class="dv">5</span>]</span>
<span id="cb44-35"><a href="#cb44-35" aria-hidden="true" tabindex="-1"></a>                              ,<span class="at">name=</span><span class="st">"cKO ac"</span>)</span>
<span id="cb44-36"><a href="#cb44-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb44-37"><a href="#cb44-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotTracks</span>(<span class="fu">list</span>(<span class="fu">IdeogramTrack</span>(<span class="fu">seqnames</span>(gr))</span>
<span id="cb44-38"><a href="#cb44-38" aria-hidden="true" tabindex="-1"></a>    ,<span class="fu">GenomeAxisTrack</span>(gr)</span>
<span id="cb44-39"><a href="#cb44-39" aria-hidden="true" tabindex="-1"></a>    ,<span class="fu">AnnotationTrack</span>(GenomicRanges<span class="sc">::</span><span class="fu">reduce</span>(<span class="fu">genes</span>(GRCm38)[goi]),</span>
<span id="cb44-40"><a href="#cb44-40" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fill=</span><span class="st">'gray'</span>,<span class="at">name=</span><span class="st">'DEG'</span>,</span>
<span id="cb44-41"><a href="#cb44-41" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fontcolor.title=</span><span class="st">'black'</span>)</span>
<span id="cb44-42"><a href="#cb44-42" aria-hidden="true" tabindex="-1"></a>    ,<span class="fu">GeneRegionTrack</span>(<span class="fu">getGeneRegionTrackForGviz</span>(GRCm38),</span>
<span id="cb44-43"><a href="#cb44-43" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transcriptAnnotation=</span><span class="st">'symbol'</span>,</span>
<span id="cb44-44"><a href="#cb44-44" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fill=</span>my_palette[<span class="dv">1</span>],<span class="at">col=</span>my_palette[<span class="dv">1</span>],<span class="at">col.line=</span><span class="cn">NULL</span>,</span>
<span id="cb44-45"><a href="#cb44-45" aria-hidden="true" tabindex="-1"></a>                    <span class="at">min.height=</span><span class="dv">4</span>,</span>
<span id="cb44-46"><a href="#cb44-46" aria-hidden="true" tabindex="-1"></a>                    <span class="at">background.title=</span>my_palette[<span class="dv">1</span>],</span>
<span id="cb44-47"><a href="#cb44-47" aria-hidden="true" tabindex="-1"></a>                    <span class="at">name=</span><span class="st">"Ensembl"</span>)</span>
<span id="cb44-48"><a href="#cb44-48" aria-hidden="true" tabindex="-1"></a>    ,k27me3_wt_track</span>
<span id="cb44-49"><a href="#cb44-49" aria-hidden="true" tabindex="-1"></a>    ,<span class="fu">AnnotationTrack</span>(K27me3_wt_peaks,<span class="at">chr=</span><span class="fu">as.character</span>(<span class="fu">seqnames</span>(gr)),</span>
<span id="cb44-50"><a href="#cb44-50" aria-hidden="true" tabindex="-1"></a>                     <span class="at">fill=</span>my_palette[<span class="dv">2</span>],<span class="at">col=</span>my_palette[<span class="dv">2</span>],<span class="at">background.title=</span>my_palette[<span class="dv">2</span>])</span>
<span id="cb44-51"><a href="#cb44-51" aria-hidden="true" tabindex="-1"></a>    ,k27me3_ko_track</span>
<span id="cb44-52"><a href="#cb44-52" aria-hidden="true" tabindex="-1"></a>    ,<span class="fu">AnnotationTrack</span>(K27me3_cko_peaks,<span class="at">chr=</span><span class="fu">as.character</span>(<span class="fu">seqnames</span>(gr)),</span>
<span id="cb44-53"><a href="#cb44-53" aria-hidden="true" tabindex="-1"></a>                     <span class="at">fill=</span>my_palette[<span class="dv">3</span>],<span class="at">col=</span>my_palette[<span class="dv">3</span>],<span class="at">background.title=</span>my_palette[<span class="dv">3</span>])</span>
<span id="cb44-54"><a href="#cb44-54" aria-hidden="true" tabindex="-1"></a>    ,k27ac_wt_track</span>
<span id="cb44-55"><a href="#cb44-55" aria-hidden="true" tabindex="-1"></a>    ,<span class="fu">AnnotationTrack</span>(K27ac_wt_peaks,<span class="at">chr=</span><span class="fu">as.character</span>(<span class="fu">seqnames</span>(gr)),</span>
<span id="cb44-56"><a href="#cb44-56" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fill=</span>my_palette[<span class="dv">4</span>],<span class="at">col=</span>my_palette[<span class="dv">4</span>],</span>
<span id="cb44-57"><a href="#cb44-57" aria-hidden="true" tabindex="-1"></a>                    <span class="at">background.title=</span>my_palette[<span class="dv">4</span>])</span>
<span id="cb44-58"><a href="#cb44-58" aria-hidden="true" tabindex="-1"></a>    ,k27ac_ko_track</span>
<span id="cb44-59"><a href="#cb44-59" aria-hidden="true" tabindex="-1"></a>    ,<span class="fu">AnnotationTrack</span>(K27ac_cko_peaks,<span class="at">chr=</span><span class="fu">as.character</span>(<span class="fu">seqnames</span>(gr)),</span>
<span id="cb44-60"><a href="#cb44-60" aria-hidden="true" tabindex="-1"></a>                     <span class="at">fill=</span>my_palette[<span class="dv">5</span>],<span class="at">col=</span>my_palette[<span class="dv">5</span>],<span class="at">background.title=</span>my_palette[<span class="dv">5</span>])</span>
<span id="cb44-61"><a href="#cb44-61" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb44-62"><a href="#cb44-62" aria-hidden="true" tabindex="-1"></a>  ,<span class="at">from=</span><span class="fu">start</span>(gr),<span class="at">to=</span><span class="fu">end</span>(gr)</span>
<span id="cb44-63"><a href="#cb44-63" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb44-64"><a href="#cb44-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-65"><a href="#cb44-65" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Now Showing:  Hoxa13 \n"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped</code></pre>
</div>
<div class="cell-output-display">
<p><img src="GViz-Diagrams_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Now Showing:  Hoxc10 \n"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped</code></pre>
</div>
<div class="cell-output-display">
<p><img src="GViz-Diagrams_files/figure-html/unnamed-chunk-4-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Now Showing:  Hoxd10 \n"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped</code></pre>
</div>
<div class="cell-output-display">
<p><img src="GViz-Diagrams_files/figure-html/unnamed-chunk-4-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Now Showing:  Wnt7a \n"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped</code></pre>
</div>
<div class="cell-output-display">
<p><img src="GViz-Diagrams_files/figure-html/unnamed-chunk-4-4.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Now Showing:  Pax6 \n"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped
Warning in normarg_seqnames2(seqnames, seqinfo): levels in 'seqnames' with no
entries in 'seqinfo' were dropped</code></pre>
</div>
<div class="cell-output-display">
<p><img src="GViz-Diagrams_files/figure-html/unnamed-chunk-4-5.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>